js/js.go

```diff
@@ -59,7 +59,12 @@ func (o *Object) New(args ...interface{}) *Object { return o.object.New(args...)
 func (o *Object) Bool() bool { return o.object.Bool() }
 
 // String returns the object converted to string according to JavaScript type conversions.
-func (o *Object) String() string { return o.object.String() }
+func (o *Object) String() string {
+	if o == Undefined {
+		return "undefined"
+	}
+	return o.object.String()
+}
```

Comment:
This should not be changed, because it is just a proxy when the JS value is wrapped by the real `*js.Object` struct (in case of being stored in an interface or `js.Object` without pointer).